subsys/bluetooth/host/gatt.c

```diff
@@ -2106,7 +2106,7 @@ static struct bt_conn *bt_gatt_ccc_cfg_conn_lookup(const struct bt_gatt_ccc_cfg
 
 	conn = bt_conn_lookup_addr_le(cfg->id, &cfg->peer);
 
-	if (bt_gatt_ccc_cfg_is_matching_conn(conn, cfg)) {
+	if (conn && bt_gatt_ccc_cfg_is_matching_conn(conn, cfg)) {
```

Comment:
this may leak conn reference...

Reply:
This should fix the issue: https://github.com/zephyrproject-rtos/zephyr/pull/59610. I'll update this PR to backport the new PR too when it get merged.

Reply:
@sjanc Could you remove your change request so https://github.com/zephyrproject-rtos/zephyr/pull/59610 can be backported. Or maybe we can just delete this PR and just backport the linked one. It may be better.